src/control/server/mgmt_svc.go

```diff
@@ -475,6 +475,40 @@ func (svc *mgmtSvc) SmdListPools(ctx context.Context, req *mgmtpb.SmdPoolReq) (*
 	return resp, nil
 }
 
+// PrepShutdown implements the method defined for the Management Service.
+//
+// Prepare data-plane instance managed by control-plane for a controlled shutdown,
+// identified by unique rank.
+func (svc *mgmtSvc) PrepShutdown(ctx context.Context, req *mgmtpb.PrepShutdownReq) (*mgmtpb.DaosResp, error) {
+	svc.log.Debugf("MgmtSvc.PrepShutdown dispatch, req:%+v\n", *req)
+
+	var mi *IOServerInstance
+	for _, i := range svc.harness.Instances() {
+		if i.hasSuperblock() && i.getSuperblock().Rank.Equals(ioserver.NewRankPtr(req.Rank)) {
+			mi = i
+			break
+		}
+	}
+
+	if mi == nil {
+		return nil, errors.Errorf("rank %d not found on this server", req.Rank)
+	}
+
+	dresp, err := mi.CallDrpc(drpc.ModuleMgmt, drpc.MethodPrepShutdown, req)
```

Comment:
Does it make sense to record that this operation was successful as an attribute on the membership entry? If I understand correctly, this method would be called via a compound operation prior to actually shutting down an instance, right? But if it fails for some reason, will that error get back to the admin? My concern is that we don't want the admin to request a shutdown which fails this step but then proceeds on to killing the instance, triggering a rebuild or other undesirable behavior.

Reply:
yes this is part of the follow-up work that will perform the fanout call over gRPC, this PR is just a plumbing for the server side to put in place to handler where the evict disable and inhibit rebuild can be called from more details in https://jira.hpdd.intel.com/browse/DAOS-2883

Reply:
I imagine we will want to store various state in the member records to enable decisions to be made based on condition of the members